src/codechandler/CodecHandler.cpp

```diff
@@ -0,0 +1,48 @@
+#include "CodecHandler.h"
+
+bool CodecHandler::GetInformation(kodi::addon::InputstreamInfo& info)
+{
+  AP4_GenericAudioSampleDescription* asd(nullptr);
+  if (m_sampleDescription)
+  {
+    if ((asd = dynamic_cast<AP4_GenericAudioSampleDescription*>(m_sampleDescription)))
+    {
+      if ((!info.GetChannels() && asd->GetChannelCount() != info.GetChannels()) ||
+          (!info.GetSampleRate() && asd->GetSampleRate() != info.GetSampleRate()) ||
+          (!info.GetBitsPerSample() && asd->GetSampleSize() != info.GetBitsPerSample()))
+      {
+        if (!info.GetChannels())
+          info.SetChannels(asd->GetChannelCount());
+        if (!info.GetSampleRate())
+          info.SetSampleRate(asd->GetSampleRate());
+        if (!info.GetBitsPerSample())
+          info.SetBitsPerSample(asd->GetSampleSize());
+        return true;
+      }
+    }
+    else
+    {
+      //Netflix Framerate
+      AP4_Atom* atom;
+      AP4_UnknownUuidAtom* nxfr;
+      static const AP4_UI08 uuid[16] = {0x4e, 0x65, 0x74, 0x66, 0x6c, 0x69, 0x78, 0x46,
+                                        0x72, 0x61, 0x6d, 0x65, 0x52, 0x61, 0x74, 0x65};
+
+      if ((atom =
+               m_sampleDescription->GetDetails().GetChild(static_cast<const AP4_UI08*>(uuid), 0)) &&
+          (nxfr = dynamic_cast<AP4_UnknownUuidAtom*>(atom)) && nxfr->GetData().GetDataSize() == 10)
+      {
+        AP4_UI16 fpsRate = nxfr->GetData().GetData()[7] | nxfr->GetData().GetData()[6] << 8;
+        AP4_UI16 fpsScale = nxfr->GetData().GetData()[9] | nxfr->GetData().GetData()[8] << 8;
```

Comment:
since we handle all this as `unsigned int` these vars should be declared as `unsigned int`